lib/jobs/ucs-catalog.js

```diff
@@ -80,19 +80,18 @@ function UcsCatalogJobFactory(
                 var username = self.ucs.settings.username;
                 var password = self.ucs.settings.password;
                 var ucs = self.ucs.settings.ucs;
-                var baseurl= self.ucs.settings.protocol+"://"+self.ucs.settings.host+ ":"+self.ucs.settings.port;/* jshint ignore:line */
-                var url= baseurl + "/catalog/?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
+                var url=  "/catalog?host=" + ucs+ "&user="+ username + "&password=" + password + "&identifier="+ node.name;/* jshint ignore:line */
                 return self.ucs.clientRequest(url);
             })
             .then(function(res) {
                 assert.object(res);
                 return res.body;
             })
-            .then(function(data) {
+            .map(function(data) {
```

Comment:
isn't the `.map` function only for arrays?  Looks like you then assert data should be an object so I'm not sure if that makes sense